### PR TITLE
find_parcellation_cut_coords 

### PR DESCRIPTION
**Pull request by amadeuskanaan**
takes care of #646

### CODE REVIEW

File: nilearn/plotting/find_cuts.py [hunk 1]
```diff
@@ -269,3 +273,59 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
     cut_coords.sort()
 
     return _transform_cut_coords(cut_coords, direction, affine)
+
+
+def find_parcellation_atlas_cut_coords(label_img):
+
```
Comment (GaelVaroquaux) [inline nilearn/plotting/find_cuts.py]: Remove empty line

File: nilearn/plotting/find_cuts.py [hunk 2]
```diff
@@ -269,3 +273,59 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
     cut_coords.sort()
 
     return _transform_cut_coords(cut_coords, direction, affine)
+
+
+def find_parcellation_atlas_cut_coords(label_img):
```
Comment (GaelVaroquaux) [inline nilearn/plotting/find_cuts.py]: We should probably add a way to specify the background label (use the same name for the argument as in NiftiLabelsMasker)

File: nilearn/plotting/find_cuts.py [hunk 3]
```diff
@@ -269,3 +273,59 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
...
+        Center coordinates for label regions
+    """
+
+    # grab data and affine
+    label_img = check_niimg(label_img)
+    label_data = label_img.get_data()
+    label_affine = label_img.get_affine()
+
+    # grab number of unique values in 3d image
+    label_unique = np.unique(label_data)[1:]
```
Comment (GaelVaroquaux) [inline nilearn/plotting/find_cuts.py]: This should be replaced by relying on the parameter that specifies the background label.

File: nilearn/plotting/find_cuts.py [hunk 4]
```diff
@@ -269,3 +273,59 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
...
+        labels, label_nb = ndimage.label(volume)
+        if label_nb == 1:
+            volume = volume.astype(np.bool)
+
+        # get parcellation center of mass
+        center_of_mass = ndimage.center_of_mass(volume)
+        world_coords = coord_transform(center_of_mass[0],
+                                                  center_of_mass[1],
+                                                  center_of_mass[2],
+                                                  label_affine)
```
Comment (GaelVaroquaux) [inline nilearn/plotting/find_cuts.py]: You'll need to align tabulations (here and below).

File: nilearn/plotting/find_cuts.py [hunk 5]
```diff
@@ -269,3 +273,59 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
...
+
+        #take the largest connected component
+        volume = np.asarray(cur_img)
+        labels, label_nb = ndimage.label(volume)
+        if label_nb == 1:
+            volume = volume.astype(np.bool)
+
+        # get parcellation center of mass
+        center_of_mass = ndimage.center_of_mass(volume)
+        world_coords = coord_transform(center_of_mass[0],
```
Comment (GaelVaroquaux) [inline nilearn/plotting/find_cuts.py]: Can you do the coord_transform at the end of the for loop.

**Comment by GaelVaroquaux** [comment]
Add documentation
=================

Add to doc/modules/reference.rst

Add a seealso in the plot_connectome docstring (see the docstring of np.std as an example).

Add to the narrative documentation where we introduce plot_connectome (in connectivity/functional_connectomes.rst)

Tests (unit tests)
===============

Add a test in plotting/tests/test_find_cuts.py that tests the behavior on synthetic data.

**Comment by amadeuskanaan** [comment]
see added background label param and out of loop coord transform

File: nilearn/plotting/find_cuts.py [hunk 6]
```diff
@@ -269,3 +273,84 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
...
+    label_list = []
+
+    for i, cur_label in enumerate(unique_labels):
+        cur_img = labels_data == cur_label
+
+        #take the largest connected component
+        volume = np.asarray(cur_img)
+        labels, label_nb = ndimage.label(volume)
+        if label_nb == 1:
+            volume = volume.astype(np.bool)
```
Comment (AlexandreAbraham) [inline nilearn/plotting/find_cuts.py]: I don't see the utility of the code here.

File: nilearn/plotting/find_cuts.py [hunk 7]
```diff
@@ -269,3 +273,84 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
...
+
+        # get parcellation center of mass
+        center_of_mass = ndimage.center_of_mass(volume)
+        center_coord = (center_of_mass[0],
+                        center_of_mass[1],
+                        center_of_mass[2])
+
+        # dump label region and coordinates into a dictionary
```
Comment (AlexandreAbraham) [inline nilearn/plotting/find_cuts.py]: Comment outdated

File: nilearn/plotting/find_cuts.py [hunk 8]
```diff
@@ -269,3 +273,84 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
...
+        label_list.append(cur_label)
+        coord_list.append((center_coord[0],
+                           center_coord[1],
+                           center_coord[2]))
```
Comment (AlexandreAbraham) [inline nilearn/plotting/find_cuts.py]: This is highly redundant with the lines above.

File: nilearn/plotting/find_cuts.py [hunk 9]
```diff
@@ -269,3 +273,84 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
...
+
+    # transform coordinates
+    world_coords = [np.asarray(coord_transform(i[0],i[1],i[2],labels_affine)) for i in coord_list]
```
Comment (AlexandreAbraham) [inline nilearn/plotting/find_cuts.py]: You can apply coord_transform in the for loop.

**Comment by amadeuskanaan** [comment]
Documentation added, though still need to add link to example in connectivity/functional_connectomes.rst.

File: nilearn/plotting/find_cuts.py [hunk 10]
```diff
@@ -269,3 +269,79 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
     cut_coords.sort()
 
     return _transform_cut_coords(cut_coords, direction, affine)
+
+
+def find_parcellation_atlas_cut_coords(labels_img, background_label=0):
+
```
Comment (banilo) [inline nilearn/plotting/find_cuts.py]: wipe empty line

File: nilearn/plotting/find_cuts.py [hunk 11]
```diff
@@ -269,3 +269,79 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
...
+    """ Grab coordinates of center of mass of 3D parcellation atlas
+
+    Parameters
+    ----------
+    labels_img: 3D Nifti1Image
+        A brain parcellation atlas.
```
Comment (banilo) [inline nilearn/plotting/find_cuts.py]: add notion that one region = one integer

Comment (banilo) [inline nilearn/plotting/find_cuts.py]: ...and that it can also be only a single target region in the 3d

File: nilearn/plotting/find_cuts.py [hunk 12]
```diff
@@ -269,3 +269,79 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
...
+    background_label: number, optional
+        Label used in labels_img to represent background.
+
+    Returns
+    -------
+    labels_list: List
+        Label region values
+
+    coords: List
+        World coordinates for center of label regions
```
Comment (banilo) [inline nilearn/plotting/find_cuts.py]: in mm/cm?

File: nilearn/plotting/find_cuts.py [hunk 13]
```diff
@@ -269,3 +269,79 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
...
+
+    """ Grab coordinates of center of mass of 3D parcellation atlas
+
+    Parameters
+    ----------
+    labels_img: 3D Nifti1Image
+        A brain parcellation atlas.
+
+    background_label: number, optional
+        Label used in labels_img to represent background.
```
Comment (banilo) [inline nilearn/plotting/find_cuts.py]: LabelS

can me more than one

File: nilearn/plotting/find_cuts.py [hunk 14]
```diff
@@ -269,3 +269,79 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
...
+        x,y,z = ndimage.center_of_mass(labels)
+
+        # dump label region and coordinates into a dictionary
+        label_list.append(cur_label)
+        coord_list.append((x,y,z))
+
+    # transform coordinates
+    coords = [coord_transform(i[0],i[1],i[2],labels_affine) for i in coord_list]
+
+    return label_list, np.array(coords)
```
Comment (banilo) [inline nilearn/plotting/find_cuts.py]: docstring said it would ne 2 lists

File: nilearn/plotting/find_cuts.py [hunk 15]
```diff
@@ -269,3 +269,79 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
...
+    unique_labels = set(np.unique(labels_data)) - set([background_label])
+
+    # grab center of mass from parcellations and dump into coords list
+    coord_list = []
+    label_list = []
+
+    for cur_label in unique_labels:
+        cur_img = labels_data == cur_label
+
+        #take the largest connected component
```
Comment (banilo) [inline nilearn/plotting/find_cuts.py]: PEP8

File: nilearn/plotting/find_cuts.py [hunk 16]
```diff
@@ -269,3 +269,79 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
...
+        volume = np.asarray(cur_img)
+        labels, label_nb = ndimage.label(volume)
+        label_count = np.bincount(labels.ravel())
+        labels ==label_count.argmax()
```
Comment (banilo) [inline nilearn/plotting/find_cuts.py]: makes no sense

File: nilearn/plotting/find_cuts.py [hunk 17]
```diff
@@ -269,3 +269,79 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
...
+        label_count[0] = 0
```
Comment (banilo) [inline nilearn/plotting/find_cuts.py]: not used anywhere

File: nilearn/plotting/find_cuts.py [hunk 18]
```diff
@@ -269,3 +269,79 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
...
+        coord_list.append((x,y,z))
+
+    # transform coordinates
+    coords = [coord_transform(i[0],i[1],i[2],labels_affine) for i in coord_list]
+
+    return label_list, np.array(coords)
+
+
+def find_probabilistic_atlas_cut_coords(label_img):
+    """ Grab coordinates of center probabolistic atlas 4D image
```
Comment (banilo) [inline nilearn/plotting/find_cuts.py]: typo + difference from above not clear

File: nilearn/plotting/find_cuts.py [hunk 19]
```diff
@@ -269,3 +269,79 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
...
+    coord_list = []
+    label_list = []
+
+    for cur_label in unique_labels:
+        cur_img = labels_data == cur_label
+
+        #take the largest connected component
+        volume = np.asarray(cur_img)
+        labels, label_nb = ndimage.label(volume)
+        label_count = np.bincount(labels.ravel())
```
Comment (bthirion) [inline nilearn/plotting/find_cuts.py]: As a side not, I'd use:
_, _, _, label_count = np.unique(labels)

File: doc/connectivity/functional_connectomes.rst [hunk 20]
```diff
@@ -208,10 +208,16 @@ with MNI coordinates for each region (see for instance example:
...
     >>> atlas_filename = datasets.fetch_msdl_atlas().maps
 
-For another atlas this information can be computed for each region with
-the :func:`nilearn.plotting.find_xyz_cut_coords` function
-(see example:
-:ref:`example_connectivity_plot_multi_subject_connectome.py`)::
+For atlases without readily available label coordinates, center coordinates
+can be computed for each region on hard parcellation or probabilistic atlases.
+
+For hard parcellation atlases (eg. :func:`datasets.fetch_atlas_destrieux_2009`), center
```
Comment (GaelVaroquaux) [inline doc/connectivity/functional_connectomes.rst]: Here, I would rather have a bullet-point list with very short formulation: something like:

- For hard parcellation atlases: use ... 

- For probabiliistic atlases, use ...

File: nilearn/plotting/find_cuts.py (hunk 19 above)
Comment (GaelVaroquaux) [inline nilearn/plotting/find_cuts.py]: Let's stay with bincount, which works on old numpy versions.

File: nilearn/plotting/find_cuts.py (hunk 17 above)
Comment (GaelVaroquaux) [inline nilearn/plotting/find_cuts.py]: Indeed, it should happen before the bincount.

File: nilearn/plotting/find_cuts.py [hunk 21]
```diff
@@ -269,3 +269,79 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
...
+        labels ==label_count.argmax()
+        label_count[0] = 0
+
+        # get parcellation center of mass
+        x,y,z = ndimage.center_of_mass(labels)
```
Comment (GaelVaroquaux) [inline nilearn/plotting/find_cuts.py]: This code is not pep8. You need spaces after comas (also present in other parts of the docs).

You should integrate a pep8 checker to your editor.

File: nilearn/plotting/find_cuts.py (hunk 14 above)
Comment (GaelVaroquaux) [inline nilearn/plotting/find_cuts.py]: Thinking about it, I would really make returning the label list optional and not returned by default (and as a consequence move it to a second return argument).

The reason being that to simply plot things in a plot_connectome, after using a NiftiLabelsMakser, people don't need that. And that's going to the 90% usecase.

File: nilearn/plotting/find_cuts.py [hunk 22]
```diff
@@ -269,3 +269,79 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
...
+    Returns
+    -------
+    coords: list
+       center coordinates for each label region of the probabilistic atlas
+    """
+
+    label_img = check_niimg(label_img)
+    label_imgs = iter_img(label_img)
+    coords = [find_xyz_cut_coords(img) for img in label_imgs]
+    return coords
```
Comment (GaelVaroquaux) [inline nilearn/plotting/find_cuts.py]: I would convert that to an array.

File: nilearn/plotting/find_cuts.py [hunk 23]
```diff
@@ -269,3 +269,79 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
...
+    """ Grab coordinates of center probabolistic atlas 4D image
+
+    Parameters
+    ----------
+    label_img: 4D Nifti1Image
+        A probabilistic brain atlas
+
+    Returns
+    -------
+    coords: list
```
Comment (GaelVaroquaux) [inline nilearn/plotting/find_cuts.py]: I would say here that it's an array of shape (3, n_maps)

File: nilearn/plotting/tests/test_find_cuts.py [hunk 24]
```diff
@@ -94,3 +94,38 @@ def test_tranform_cut_coords():
...
+def test_find_parcellation_atlas_cut_coords():
+    data= np.zeros((200,200,200))
+    x_map_a, y_map_a, z_map_a = 40, 40, 40
+    x_map_b, y_map_b, z_map_b = 100, 100 , 100
+    data[x_map_a - 10:x_map_a+10, y_map_a - 4 :y_map_a + 4    , z_map_a-20 : z_map_a+20] = 1
+    data[x_map_b - 20:x_map_b+20, y_map_b - 8 :y_map_b + 8    , z_map_b-50 : z_map_b+50] = 2
+
+    # identity affine
+    affine = np.eye(4)
+    img=nibabel.Nifti1Image(data,affine)
```
Comment (GaelVaroquaux) [inline nilearn/plotting/tests/test_find_cuts.py]: PEP8. You are missing a white space.

File: nilearn/plotting/find_cuts.py [hunk 25]
```diff
@@ -269,3 +269,79 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
     cut_coords.sort()
 
     return _transform_cut_coords(cut_coords, direction, affine)
+
+
+def find_parcellation_atlas_cut_coords(labels_img, background_label=0):
+
+    """ Grab coordinates of center of mass of 3D parcellation atlas
```
Comment (GaelVaroquaux) [inline nilearn/plotting/find_cuts.py]: Rather than "Grab", I would use "Return"

File: nilearn/plotting/find_cuts.py [hunk 26]
```diff
@@ -269,3 +269,79 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
     cut_coords.sort()
 
     return _transform_cut_coords(cut_coords, direction, affine)
+
+
+def find_parcellation_atlas_cut_coords(labels_img, background_label=0):
```
Comment (GaelVaroquaux) [inline nilearn/plotting/find_cuts.py]: I would call this function "find_parcellation_cut_coords" (ie: no atlas)

File: nilearn/plotting/find_cuts.py (hunk 10 above)
Comment (amadeuskanaan) [inline nilearn/plotting/find_cuts.py]: done

File: nilearn/plotting/find_cuts.py (hunk 11 above)
Comment (amadeuskanaan) [inline nilearn/plotting/find_cuts.py]: done..

File: nilearn/plotting/find_cuts.py (hunk 12 above)
Comment (amadeuskanaan) [inline nilearn/plotting/find_cuts.py]: mm

Comment (banilo) [inline nilearn/plotting/find_cuts.py]: You could add that in the description here

Comment (amadeuskanaan) [inline nilearn/plotting/find_cuts.py]: coming with next commit.

File: nilearn/plotting/find_cuts.py (hunk 22 above)
Comment (amadeuskanaan) [inline nilearn/plotting/find_cuts.py]: done

File: nilearn/plotting/find_cuts.py (hunk 23 above)
Comment (amadeuskanaan) [inline nilearn/plotting/find_cuts.py]: okay.

File: nilearn/plotting/tests/test_find_cuts.py (hunk 24 above)
Comment (amadeuskanaan) [inline nilearn/plotting/tests/test_find_cuts.py]: i have pep8...will turn automatic formatting on.

File: nilearn/plotting/find_cuts.py (hunk 25 above)
Comment (amadeuskanaan) [inline nilearn/plotting/find_cuts.py]: done

File: nilearn/plotting/find_cuts.py (hunk 26 above)
Comment (amadeuskanaan) [inline nilearn/plotting/find_cuts.py]: i guess same goes for the other one ---> 'find_probabilistic_cut_coords'.

Comment (GaelVaroquaux) [inline nilearn/plotting/find_cuts.py]: > > +def find_parcellation_atlas_cut_coords(labels_img, background_label=0):

> i guess same goes for the other one ---> 'find_probabilistic_cut_coords'.

No: it 'a parcellation', and a 'probabilistic atlas'

Comment (amadeuskanaan) [inline nilearn/plotting/find_cuts.py]: noted.....

File: nilearn/plotting/find_cuts.py (hunk 14 above)
Comment (amadeuskanaan) [inline nilearn/plotting/find_cuts.py]: done. with next commit.

File: nilearn/plotting/find_cuts.py (hunk 18 above)
Comment (amadeuskanaan) [inline nilearn/plotting/find_cuts.py]: typo. u got me. 
this func is to extract coords from 4d vols with fuzzy maps in each 3d (eg. msdl)

File: doc/connectivity/functional_connectomes.rst (hunk 20 above)
Comment (amadeuskanaan) [inline doc/connectivity/functional_connectomes.rst]: added.

**Comment by amadeuskanaan** [comment]
tests added

File: doc/connectivity/functional_connectomes.rst [hunk 27]
```diff
@@ -208,10 +208,14 @@ with MNI coordinates for each region (see for instance example:
...
 
-For another atlas this information can be computed for each region with
-the :func:`nilearn.plotting.find_xyz_cut_coords` function
-(see example:
-:ref:`example_connectivity_plot_multi_subject_connectome.py`)::
+For atlases without readily available label coordinates, center coordinates
+can be computed for each region on hard parcellation or probabilistic atlases.
+
+  * For hard parcellation atlases (eg. :func:`datasets.fetch_atlas_destrieux_2009`),
+  use the :func:`nilearn.plotting.find_parcellation_atlas_cut_coords` function.
```
Comment (GaelVaroquaux) [inline doc/connectivity/functional_connectomes.rst]: The indentation of this line is wrong (same thing below): it needs to be indented with the same indentation as the 'F' of the 'For' above.

Comment (amadeuskanaan) [inline doc/connectivity/functional_connectomes.rst]: done

File: doc/connectivity/functional_connectomes.rst [hunk 28]
```diff
@@ -208,10 +208,14 @@ with MNI coordinates for each region (see for instance example:
...
-For another atlas this information can be computed for each region with
-the :func:`nilearn.plotting.find_xyz_cut_coords` function
-(see example:
-:ref:`example_connectivity_plot_multi_subject_connectome.py`)::
+For atlases without readily available label coordinates, center coordinates
+can be computed for each region on hard parcellation or probabilistic atlases.
+
+  * For hard parcellation atlases (eg. :func:`datasets.fetch_atlas_destrieux_2009`),
+    use the :func:`nilearn.plotting.find_parcellation_atlas_cut_coords` function.
+  * For probabilistic atlases, use the :func:`nilearn.plotting.find_probabilistic_atlas_cut_coords`
```
Comment (AlexandreAbraham) [inline doc/connectivity/functional_connectomes.rst]: MSDL is an example of probabilistic atlas.

File: nilearn/plotting/find_cuts.py [hunk 29]
```diff
@@ -74,7 +75,7 @@ def find_xyz_cut_coords(img, mask=None, activation_threshold=None):
         return .5 * np.array(data.shape)
     if activation_threshold is None:
         activation_threshold = fast_abs_percentile(my_map[my_map != 0].ravel(),
-                                                   80)
+            80)
```
Comment (AlexandreAbraham) [inline nilearn/plotting/find_cuts.py]: PEP8

File: nilearn/plotting/find_cuts.py [hunk 30]
```diff
@@ -219,7 +220,7 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
     for _ in range(n_cuts):
         # Find a peak
         max_along_axis = np.unravel_index(np.abs(data).argmax(),
-                                          data.shape)[axis]
+            data.shape)[axis]
```
Comment (AlexandreAbraham) [inline nilearn/plotting/find_cuts.py]: PEP8

**Comment by AlexandreAbraham** [comment]
LGTM

**Comment by GaelVaroquaux** [comment]
The situation with the PR is the following:

On a non lateralized atlas, the centers detected appear randomly on the left and the right. As these atlases are very common, this means that we need to (optionally) choose centers only in one hemisphere.

There are 3 cases:

1. A label only present in one hemisphere (lateralized region)

2. A label at the center

3. A label present in 2 different regions (as in the Yeo or HO atlas).

For a given label, here is the logic I propose (which should be optional, but turned on by default):

<pre>
1. Find the medial plane (requires inversing the affine and applying it to (0, 0, 0)
2. if the weight of the label is fully on one side, we are in case 1, just use the current code
3. elif the largest CC is on both sides, we are on case 2, just use the current code
4. else take the largest CC on one hemisphere (left by default, but user-settable).
</pre>

**Comment by AlexandreAbraham** [comment]
What is the status of this PR? I'd like not to let it rot too long...

File: nilearn/plotting/tests/test_find_cuts.py [hunk 31]
```diff
@@ -94,3 +95,69 @@ def test_tranform_cut_coords():
     for direction in 'xyz':
         assert_equal(len(_transform_cut_coords(cut_coords, direction, affine)),
                      n_cuts)
+
+
+def test_find_parcellation_atlas_cut_coords():
+    data = np.zeros((250, 250, 250))
```
Comment (bthirion) [inline nilearn/plotting/tests/test_find_cuts.py]: You use relatively large numbers. hlaving them or even dividing by 4 or 5 would save time.

File: nilearn/plotting/tests/test_find_cuts.py [hunk 32]
```diff
@@ -94,3 +95,69 @@ def test_tranform_cut_coords():
...
+def test_find_probabilistic_atlas_cut_coords():
+    # make data
+    arr1 = np.zeros((200, 200, 200))
+    x_map_a, y_map_a, z_map_a = 30, 60, 120
+    arr1[x_map_a - 10:x_map_a + 10, y_map_a - 20:y_map_a + 20, z_map_a - 30: z_map_a + 30] = 1
+
+    arr2 = np.zeros((200, 200, 200))
+    x_map_b, y_map_b, z_map_b = 40, 50, 60
+    arr2[x_map_b - 10:x_map_b + 10, y_map_b - 20:y_map_b + 20, z_map_b - 30: z_map_b + 30] = 1
+
```
Comment (bthirion) [inline nilearn/plotting/tests/test_find_cuts.py]: Same comment here

**Comment by bthirion** [comment]
The PR looks mostly good. The CircleCI failure does not seem to be related to this PR...

**Comment by lesteve** [comment]
You need to rebase on master and fix the merge conflicts on this one. @amadeuskanaan do give us a shout if you need some help.

> The PR looks mostly good. The CircleCI failure does not seem to be related to this PR...

Yeah this is an old CircleCI failure that has been fixed during the sprint in master. I guess what happened is that this PR branch was branched off master before the fix.

**Comment by amadeuskanaan** [comment]
a bit backed up with time sensitive stuff at the moment.. will get into again this week and ping again in a few days.

**Comment by KamalakerDadi** [comment]
Hi @amadeuskanaan, Any progress on this ?

**Comment by KamalakerDadi** [comment]
Closing as superseded with #1602

**Pull request closed without merging**